cli/azd/pkg/environment/environment.go

```diff
@@ -227,7 +228,11 @@ func (e *Environment) Reload() error {
 	}
 
 	if e.GetSubscriptionId() != "" {
-		tracing.SetGlobalAttributes(fields.SubscriptionIdKey.String(e.GetSubscriptionId()))
+		if _, err := uuid.Parse(e.GetSubscriptionId()); err == nil {
+			tracing.SetGlobalAttributes(fields.SubscriptionIdKey.String(e.GetSubscriptionId()))
+		} else {
+			tracing.SetGlobalAttributes(fields.StringHashed(fields.SubscriptionIdKey, e.GetSubscriptionId()))
+		}
```

Comment:
What is the reason that we are tracing this differently if it is a valid UUID or not? Is there a bug reported? Subscription ID always need to be an actual UUID - Should we have additional validation in all areas where subscription id is set?